Signal/src/ViewControllers/ConversationView/ConversationViewModel.m

```diff
@@ -259,6 +247,19 @@ - (void)configure
     if (![self reloadViewItems]) {
         OWSFailDebug(@"failed to reload view items in configureForThread.");
     }
+
+    [[NSNotificationCenter defaultCenter] addObserver:self
```

Comment:
Unlike the old CVC, the CVM builds a mapping immediately and continually updates it to reflect all db modifications.  Consequently, we need to start observing DB modifications notification immediately.  Previous to this PR we were vulnerable to a race wherein a DB modification occurred between the view/VM being configured and the view loading.

Reply:
I was thinking about this over the weekend and I think this race might have existed in the pre-CVM CVC.  

There's a number of ways in which these changes _might_ significantly reduce our crashes in production.